src/main/java/org/jenkinsci/plugins/envinject/EnvInjectPluginConfiguration.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.envinject;
+
+import com.google.common.annotations.VisibleForTesting;
+import hudson.Extension;
+import hudson.XmlFile;
+import java.io.File;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import static org.jenkinsci.plugins.envinject.EnvInjectPlugin.getJenkinsInstance;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+/**
+ * Configuration of {@link EnvInjectPlugin}.
+ * @author Oleg Nenashev
+ * @since 1.92
+ */
+@Extension
+public class EnvInjectPluginConfiguration extends GlobalConfiguration {
+    
+    private static final EnvInjectPluginConfiguration DEFAULT = 
+            new EnvInjectPluginConfiguration(false, false);
+            
+    private boolean hideInjectedVars;
+
+    private boolean enablePermissions;
+
+    public EnvInjectPluginConfiguration() {
+        load();
+    }
+    
+    @DataBoundConstructor
+    public EnvInjectPluginConfiguration(boolean hideInjectedVars, boolean enablePermissions) {
+        this.hideInjectedVars = hideInjectedVars;
+        this.enablePermissions = enablePermissions;
+    }
+
+    public boolean isHideInjectedVars() {
+        return hideInjectedVars;
+    }
+
+    public boolean isEnablePermissions() {
+        return enablePermissions;
+    }
+ 
+    /**
+     * Gets the default configuration of {@link EnvInjectPlugin}
+     * @return Default configuration
+     */
+    public static final @Nonnull EnvInjectPluginConfiguration getDefault() {
+        return DEFAULT;
+    }
+
+    @Override
+    protected XmlFile getConfigFile() {
+        return new XmlFile(Jenkins.XSTREAM, new File(getJenkinsInstance().getRootDir(), 
+                "envinject-plugin-configuration.xml"));
```

Comment:
IIRC @lanwen was against the approach with a full package/class name. I also agree with it since we have "envinject.xml". It makes sense to merge configurations to a common name later in envinject-2.0